tools/fs_validation.py

```diff
@@ -0,0 +1,203 @@
+    column_list = []
+    for col_name in (df_1.columns)[:5]:
+        for word in words:
+            if word in col_name and col_name in (df_2.columns):
+                column_list.append(col_name)
+    return column_list[:2]
+
+
+# COMMAND ----------
+
+# MAGIC %md Main function
+
+# COMMAND ----------
+
+
+def validate_data_between_two_sources(
+    df_name: str,
+    df_path_old_pipeline: str,
+    coef: float = 0.1,
+    limit_of_rows: int = 10000,
+) -> tuple():
+    """
+    Function that should be able to validate if there are any discrepancies between last
+     two versions of calculated Feature Store rows - bigger than specific thrash-hold
+
+    Parameters:
+    df_name: Name of table that is specified in config file
+    coef: Thrash-hold under which we expect that rows are calculated just fine
+    """
+
+    coef = 0.1 if coef is None else coef  # handle 'None' input for coef
+
+    # load table using YAML
+    if "." not in df_name:
+        df_path = get_value_from_yaml("paths", df_name)
+
+        df_1 = spark.read.format("delta").load(
+            df_path
+        )  # read "current" ODAP pipeline results
+        df_2 = spark.read.format("delta").load(
+            df_path_old_pipeline
+        )  # read "old" pipeline results
+
+        columns_to_validate = _get_columns_to_compare_non_fs(df_1)
+
+    # loading table from table catalog
+    else:
+        df_1 = spark.read.table(df_name)  # read "current" ODAP pipeline results
+        df_2 = spark.read.format("delta").load(
+            df_path_old_pipeline
+        )  # read "old" pipeline results
+
+        columns_to_validate = _get_columns_to_compare(df_1, df_2)
+
+    # get random sample of last version table
+    if limit_of_rows is None:
+        df_1_s = df_1.sample(False, coef)
+    else:
+        df_1_s = df_1.sample(False, coef).limit(limit_of_rows)
+
+    df_join = (
+        df_1_s.alias("df_1")
+        .join(df_2.alias("df_2"), on="user_id", how="inner")
+        .withColumn("diff", F.lit(0))
+    )  # add column to check validity and for further filtering
+
+    df_diff = df_join.withColumn("diff", F.lit(0))
+
+    for col in columns_to_validate:
+        df_diff = df_diff.withColumn(
+            "diff",
+            F.when(
+                F.abs(
+                    (F.col(f"df_1.{col}") - F.col(f"df_2.{col}")) / F.col(f"df_1.{col}")
+                )
+                < F.lit(coef),
+                F.col("diff"),
+            ).otherwise(F.lit(1)),
+        )
+
+    print(
+        f"There are totally {df_diff.count()} columns, {df_1.count()} from df_1 and {df_2.count()} from df_2"
+    )
+
+    print(
+        f"There were comparison done for {df_diff.count()} columns, {df_1_s.count()} from df_1 and {df_2.count()} from df_2"
+    )
+
+    df_diff = df_diff.filter(F.col("diff") == F.lit(1))
+
+    print(
+        f"There left {df_diff.count()} columns for comparison after removing valid non-different (valid) rows"
```

Comment:
```suggestion
        f"There is {df_diff.count()} columns left for comparison after removing valid non-different rows"
```
Or did the second valid convey anything important?